core/drivers/crypto/versal/ecc.c

```diff
@@ -338,32 +121,63 @@ static TEE_Result do_shared_secret(struct drvcrypt_secret_data *sdata)
 
 static TEE_Result do_sign(struct drvcrypt_sign_data *sdata)
 {
-	return sign(sdata->algo,
-		    sdata->key,
-		    sdata->message.data,
-		    sdata->message.length,
-		    sdata->signature.data,
-		    &sdata->signature.length);
+	TEE_Result ret;
+
+	ret = versal_ecc_sign(sdata->algo,
+			      sdata->key,
+			      sdata->message.data,
+			      sdata->message.length,
+			      sdata->signature.data,
+			      &sdata->signature.length);
+
+	if (ret == TEE_ERROR_NOT_SUPPORTED)
+		/* Fallback to software */
+		return pair_ops->sign(sdata->algo, sdata->key,
+				      sdata->message.data,
+				      sdata->message.length,
+				      sdata->signature.data,
+				      &sdata->signature.length);
+
+	return ret;
 }
 
 static TEE_Result do_verify(struct drvcrypt_sign_data *sdata)
 {
-	return verify(sdata->algo,
-		      sdata->key,
-		      sdata->message.data,
-		      sdata->message.length,
-		      sdata->signature.data,
-		      sdata->signature.length);
+	TEE_Result ret;
+
+	ret = versal_ecc_verify(sdata->algo,
+				sdata->key,
+				sdata->message.data,
+				sdata->message.length,
+				sdata->signature.data,
+				sdata->signature.length);
+
+	if (ret == TEE_ERROR_NOT_SUPPORTED)
+		/* Fallback to software */
+		return pub_ops->verify(sdata->algo, sdata->key,
+				       sdata->message.data,
+				       sdata->message.length,
+				       sdata->signature.data,
+				       sdata->signature.length);
```

Comment:
add braces : `if (ret == TEE_ERROR_NOT_SUPPORTED) { ... }`

Ditto at line 133

Reply:
Fixed in fixup commit a9df558c8